modules/test/playwright/tests/roles-admin-web/accountRoles.spec.ts

```diff
@@ -0,0 +1,161 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {expect, mergeTests} from '@playwright/test';
+
+import {accountsPagesTest} from '../../fixtures/accountsPagesTest';
+import {applicationsMenuPageTest} from '../../fixtures/applicationsMenuPageTest';
+import {dataApiHelpersTest} from '../../fixtures/dataApiHelpersTest';
+import {loginTest} from '../../fixtures/loginTest';
+import getRandomString from '../../utils/getRandomString';
+import performLogin, {performLogout, userData} from '../../utils/performLogin';
+import {PORTLET_URLS} from '../../utils/portletUrls';
+
+export const test = mergeTests(
+	accountsPagesTest,
+	applicationsMenuPageTest,
+	dataApiHelpersTest,
+	loginTest()
+);
+
+test('LPD-44889 Can view account roles with permissions', async ({
+	accountRolesPage,
+	accountsPage,
+	apiHelpers,
+	page,
+}) => {
+	const userAccount = await apiHelpers.headlessAdminUser.postUserAccount();
+
+	userData[userAccount.alternateName] = {
+		name: userAccount.givenName,
+		password: 'test',
+		surname: userAccount.familyName,
+	};
+
+	const companyId = await page.evaluate(() => {
+		return Liferay.ThemeDisplay.getCompanyId();
+	});
+
+	const accountAdminRole = await apiHelpers.headlessAdminUser.postRole({
+		name: getRandomString(),
+		rolePermissions: [
+			{
+				actionIds: ['ACCESS_IN_CONTROL_PANEL'],
+				primaryKey: companyId,
+				resourceName:
+					'com_liferay_account_admin_web_internal_portlet_AccountEntriesAdminPortlet',
+				scope: 1,
+			},
+		],
+	});
+
+	await apiHelpers.headlessAdminUser.assignUserToRole(
+		accountAdminRole.externalReferenceCode,
+		userAccount.id
+	);
+
+	const account1 = await apiHelpers.headlessAdminUser.postAccount({
+		name: 'Account1',
+		type: 'business',
+	});
+
+	apiHelpers.data.push({id: account1.id, type: 'account'});
+
+	const account2 = await apiHelpers.headlessAdminUser.postAccount({
+		name: 'Account2',
+		type: 'business',
+	});
+
+	apiHelpers.data.push({id: account2.id, type: 'account'});
+
+	await apiHelpers.headlessAdminUser.assignUserToAccountByEmailAddress(
+		account1.id,
+		[userAccount.emailAddress]
+	);
+	await apiHelpers.headlessAdminUser.assignUserToAccountByEmailAddress(
+		account2.id,
+		[userAccount.emailAddress]
+	);
+
+	const accountRole1 = await apiHelpers.headlessAdminUser.postRole({
+		name: getRandomString(),
+		rolePermissions: [
+			{
+				actionIds: ['UPDATE', 'VIEW_ACCOUNT_ROLES'],
+				primaryKey: '0',
+				resourceName: 'com.liferay.account.model.AccountEntry',
+				scope: 3,
+			},
+			{
+				actionIds: ['VIEW'],
+				primaryKey: '0',
+				resourceName: 'com.liferay.account.model.AccountRole',
+				scope: 3,
+			},
+		],
+		roleType: 'account',
+	});
+
+	let rolesResponse = await apiHelpers.headlessAdminUser.getAccountRoles(
+		account1.id
+	);
+
+	let role = rolesResponse?.items?.filter(
```

Comment:
Why is this step necessary? Why not just use accountRole1. Probably not a mistake, I'm just curious :)

Reply:
Yes, I think accountRole1 should role contains the id you need

Reply:
@smottal @matyaswollner I think you are right guys, I used CTRL+C/CTRL-V here and did not pay attention apparently.

Reply:
Okay, so I did some testing and apparently it does not work if I delete these what you highlighted. Kinda strange...